esmvaltool/cmor/table.py

```diff
@@ -111,6 +111,12 @@ def _load_table(self, json_file):
                 self._assign_dimensions(var, generic_levels)
                 table[var_name] = var
                 self.var_to_freq[table.name][var_name] = var.frequency
+
+            if not table.frequency:
+                from collections import Counter
+                var_freqs = (var.frequency for var in table.values())
+                table_freq, _ = Counter(var_freqs).most_common(1)[0]
+                table.frequency = table_freq
```

Comment:
Boss! I like it - I tested with a whole bunch of tables and the closest one I got was this one:
```
Counter(freqs).most_common(2)
[(' "yr"', 10), (' "yrPt"', 9)]
```
for `Eyr`